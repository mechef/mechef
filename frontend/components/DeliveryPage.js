// @flow

import React from 'react';
import Rx from 'rxjs/Rx';

import { connect } from '../state/RxState';
import deliveryActions from '../actions/deliveryActions';
import errorActions from '../actions/errorActions';
import globalActions from '../actions/globalActions';
import ErrorModal from './ErrorModal';
import DeliveryList from './DeliveryList';
import DeliveryEdit from './DeliveryEdit';
import { MeetupObject } from '../utils/flowTypes';

type Props = {
  delivery: {
    meetupList: Array<MeetupObject>,
    currentMeetupId: string,
    currentShippingId: string,
  },
  fetchDelivery$: any => Rx.Observable,
  createMeetup$: (meetup: MeetupObject) => Rx.Observable,
  updateMeetup$: (meetup: MeetupObject) => Rx.Observable,
  deleteMeetup$: () => Rx.Observable,
  setCurrentMeetupId$: (meetupId: string) => Rx.Observable,
  setError$: ({ isShowModal: boolean, title: string, message: string }) => Rx.Observable,
  error: {
    title: string,
    message: string,
    isShowModal: bool,
  },
  global: {
    backArrow: {
      isShow: boolean,
      title: string,
    },
  },
  toggleBackArrow$: string => Rx.Observable,
}

export class DeliveryPage extends React.Component<Props> {
  componentDidMount() {
    this.props.fetchDelivery$();
  }
  render() {
    const {
      delivery: { meetupList, currentMeetupId },
      setError$, error,
      global: { backArrow },
      toggleBackArrow$,
      createMeetup$,
      updateMeetup$,
      deleteMeetup$,
      setCurrentMeetupId$,
    } = this.props;
    return (
      <div className="container">
        {
          error.isShowModal ?
            <ErrorModal
              title={error.title}
              message={error.message}
              onCancel={() => setError$({ isShowModal: false, title: '', message: '' })}
            />
            : null
        }
        {
          backArrow.isShow ?
            <DeliveryEdit
              meetupList={meetupList}
              currentMeetupId={currentMeetupId}
              onCreateMeetup={createMeetup$}
              onUpdateMeetup={updateMeetup$}
              onDeleteMeetup={deleteMeetup$}
              goBack={() => toggleBackArrow$('')}
            />
            :
            <DeliveryList
              meetupList={meetupList}
              onEditDelivery={(meetupId) => {
                setCurrentMeetupId$(meetupId);
                toggleBackArrow$('Edit Delivery');
              }}
            />
        }
        <style jsx>
          {`
            .container {
              margin: 0;
              padding-top: 49px
              padding-left: 19px;
              width: 100%;
              height: 998px;
              background-color: #f8f7f7;
            }
          `}
        </style>
      </div>
    );
  }
}


const stateSelector = ({ delivery, error, global }) => ({ delivery, error, global });

const actionSubjects = {
  ...errorActions,
  ...deliveryActions,
  ...globalActions,
};

export default connect(stateSelector, actionSubjects)(DeliveryPage);
