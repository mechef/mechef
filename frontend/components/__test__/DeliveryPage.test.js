import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';

import { DeliveryPage } from '../DeliveryPage';

jest.mock('react-geosuggest', () => '[MOCK]Geosuggest');

describe('Pages With Snapshot Testing', () => {
  it('snapshot DeliveryPage Component', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <DeliveryPage
        delivery={{
          meetupList: [
            {
              _id: '59e5f7ffd48173ad48d8c8ab',
              note: '9F caferia',
              meetupEndTime: '',
              meetupStartTime: '',
              meetupSaturday: false,
              meetupFriday: false,
              meetupThursday: false,
              meetupWednesday: false,
              meetupTuesday: false,
              meetupMonday: false,
              meetupSunday: false,
              meetupLongitude: 20,
              meetupLatitude: 120,
              meetupAddress: 'Rakuten Crimson House',
              type: 'meetup',
            },
          ],
          currentMeetupId: '59e5f7ffd48173ad48d8c8ab',
          currentShippingId: '',
        }}
        fetchDelivery$={() => {}}
        createMeetup$={() => {}}
        updateMeetup$={() => {}}
        deleteMeetup$={() => {}}
        setCurrentMeetupId$={() => {}}
        setLoading$={() => {}}
        setError$={() => {}}
        toggleBackArrow$={() => {}}
        error={{
          isShowModal: false,
        }}
        global={{
          backArrow: {
            isShow: false,
          },
        }}
        t={() => {}}
      />,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
