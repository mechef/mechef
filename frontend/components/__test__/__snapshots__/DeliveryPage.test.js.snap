// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot DeliveryPage Component 1`] = `
<div
  className="jsx-4065268649 container"
>
  <div
    className="jsx-2767775796"
  >
    <div
      className="jsx-2767775796 header"
    >
      <span
        className="jsx-2767775796 title"
      >
        Delivery List
      </span>
      <div
        className="jsx-2767775796 addButton"
        onClick={[Function]}
        role="link"
        tabIndex="-1"
      >
        <i
          aria-hidden="true"
          className="jsx-2767775796 fa fa-plus plus-icon"
        />
      </div>
    </div>
    <div
      className="jsx-2767775796 delivery-list"
    >
      <div
        className="jsx-2767775796 delivery-item"
      >
        <div
          className="jsx-2767775796 delivery-content"
        >
          <p
            className="jsx-2767775796 delivery-title"
          >
            Rakuten Crimson House
          </p>
          <p
            className="jsx-2767775796 delivery-detail"
          >
            <span
              className="jsx-2767775796 delivery-subtext"
            >
              Shipping Cost:
            </span>
          </p>
        </div>
        <span
          className="jsx-2767775796 update-button"
          onClick={[Function]}
          role="button"
          tabIndex="-1"
        >
          <span
            className="jsx-2767775796 update-button-text"
          >
            UPDATE
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
